sympy/polys/multivariate_resultants.py

```diff
@@ -0,0 +1,389 @@
+"""
+This module contains functions for two multivariate resultants. These are:
+
+- Dixon's resultant.
+- Maucalay's resultant.
+
+Multivariate resultants are used to identify whether a multivariate system has
+common roots. That is when the resultant is equal to zero.
+"""
+
+from sympy import IndexedBase, Matrix, Mul, Poly, prod, fraction
+from sympy.core.compatibility import range
+from sympy.polys.monomials import monomial_deg
+from sympy.polys.monomials import itermonomials
+from sympy.polys.orderings import monomial_key
+from sympy.polys.polytools import poly_from_expr
+from sympy.functions.combinatorial.factorials import binomial
+
+class DixonResultant():
+    """
+    A class for retrieving the Dixon's resultant of a multivariate system.
+
+    Examples
+    ========
+    >>> from sympy.core import symbols
+
+    >>> from sympy.polys.multivariate_resultants import DixonResultant
+    >>> x, y = symbols('x, y')
+
+    >>> p = x + y
+    >>> q = x ** 2 + y ** 3
+    >>> h = x ** 2 + y
+
+    >>> dixon = DixonResultant(variables=[x, y], polynomials=[p, q, h])
+    >>> poly = dixon.get_dixon_polynomial()
+    >>> matrix = dixon.get_dixon_matrix(polynomial=poly)
+    >>> matrix
+    Matrix([
+    [ 0,  0, -1,  0, -1],
+    [ 0, -1,  0, -1,  0],
+    [-1,  0,  1,  0,  0],
+    [ 0, -1,  0,  0,  1],
+    [-1,  0,  0,  1,  0]])
+    >>> matrix.det()
+    0
+
+    Reference
+    ==========
+    1. [Kapur1994]_
+    2. [Palancz08]_
+
+    See Also
+    ========
+    Notebook in examples: sympy/example/notebooks.
+    """
+
+    def __init__(self, polynomials, variables):
+        """
+        A class that takes two lists, a list of polynomials and list of
+        variables. Returns the Dixon matrix of the multivariate system.
+
+        Parameters
+        ----------
+        polynomials : list of polynomials
+            A  list of m n-degree polynomials
+        variables: list
+            A list of all n variables
+        """
+        self.polynomials = polynomials
+        self.variables = variables
+
+        self.n = len(self.variables)
+        self.m = len(self.polynomials)
+
+        self.dummy_variables = self.get_dummy_variables()
```

Comment:
Would it be possible to set `dummy_variables` without calling `get_dummy_variables`. It seems that, once it has been set, there is no need for `get_dummy_variables`.